src/search/search-executor/search-generators.ts

```diff
@@ -121,6 +130,46 @@ async function generateFromQuery(input: FlowrAnalysisProvider, args: {
 	}))) as unknown as FlowrSearchElements<ParentInformation, FlowrSearchElement<ParentInformation>[]>;
 }
 
+async function generateFromTreeSitterQuery(input: FlowrAnalysisProvider, args: { source: string, captures: string[] } ): Promise<FlowrSearchElements<ParentInformation, FlowrSearchElement<ParentInformation>[]>> {
+	// if the user didn't specify a specific capture, we want to capture the outermost item
+	if(!args.captures?.length) {
+		const defaultCaptureName = 'defaultCapture';
+		args.source += ` @${defaultCaptureName}`;
+		args.captures = [defaultCaptureName];
+	}
+	// allow specifying capture names with or without the @ in front :)
+	const captures = new Set<string>(args.captures.map(c => c.startsWith('@') ? c.substring(1) : c));
+
+	const result = await input.parserInformation().treeSitterQuery(args.source);
+	const relevant = result.filter(c => captures.has(c.name));
```

Comment:
is there a way to retrieve the available captures in the query and warn if they do not match?